src/fairseq2/nn/transformer/multihead_attention.py

```diff
@@ -231,10 +237,33 @@ def __init__(
         """
         super().__init__(model_dim, num_heads)
 
+        if num_key_value_heads is None:
+            self.num_key_value_heads = num_heads
+        else:
+            self.num_key_value_heads = num_key_value_heads
+            if num_heads % num_key_value_heads != 0:
+                raise ValueError(
+                    f"`num_heads` must be divisible by `num_key_value_heads` ({num_key_value_heads}), but is {num_heads} instead."
+                )
+        self.head_dim = model_dim // num_heads
+        self.kv_repeats = self.num_heads // self.num_key_value_heads
```

Comment:
It is subjective, but I liked your original variable name `num_key_value_groups` better.